uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -300,7 +343,10 @@ define([
             return loadStatuses();
           })
           .then(function () {
-            return $q.all[initUserRole(), initOpenMode()];
+            return $q.all[
```

Comment:
reduced it further and removed few unnecessary promises, there is no related set of functions inside it to separate them out, its more of a sequence of function calls to initialize so in that sense its not very large